src/darksend.cpp

```diff
@@ -2009,22 +2032,26 @@ bool CDarksendPool::IsOutputsCompatibleWithSessionDenom(const std::vector<CTxDSO
     return true;
 }
 
-bool CDarksendPool::IsDenomCompatibleWithSession(int nDenom, CTransaction txCollateral, int& nErrorID)
+bool CDarksendPool::IsDenomCompatibleWithSession(int nDenom, CTransaction txCollateral, PoolMessage& nMessageIDRet)
 {
-    if(nDenom == 0) return false;
+    if(nDenom == 0) {
+        nMessageIDRet = ERR_DENOM;
+        return false;
+    }
 
     LogPrintf("CDarksendPool::IsDenomCompatibleWithSession -- nSessionDenom: %d nSessionUsers: %d\n", nSessionDenom, nSessionUsers);
 
     if(!fUnitTest && !IsCollateralValid(txCollateral)) {
         LogPrint("privatesend", "CDarksendPool::IsDenomCompatibleWithSession -- collateral not valid!\n");
-        nErrorID = ERR_INVALID_COLLATERAL;
+        nMessageIDRet = ERR_INVALID_COLLATERAL;
         return false;
     }
 
     if(nSessionUsers < 0) nSessionUsers = 0;
 
     if(nSessionUsers == 0) {
-        nSessionID = 1 + GetRand(999999);
+        nMessageIDRet = MSG_NOERR;
+        nSessionID = GetInsecureRand(999999)+1;
```

Comment:
Yep, we just need some random-ish number here main requirement to which is that it is not 0. It's just a session id, it doesn't reveal any information and we don't really care if it's off because it's used once and then same client will not mix on this mn for quite a long time. I doubt cryptographically secure function is needed here.

As for thread safety I think we can refactor all of GetInsecureRand/insecure_rand usage in our modules to use thread safe version later in following PRs, should be ok for now.